jwst/resample/resample_spec.py

```diff
@@ -301,11 +309,17 @@ def build_interpolated_output_wcs(self, refmodel=None):
             x_size = len(x_tan_array)
 
         # define the output wcs
-        transform = mapping | (pix_to_xtan & pix_to_ytan | undist2sky) & pix_to_wavelength
+        if model.meta.wcs.output_frame.name == 'world':
+            transform = mapping | (pix_to_xtan & pix_to_ytan | undist2sky) & pix_to_wavelength
+        else:
+            transform = mapping | (pix_to_xtan & pix_to_ytan) & pix_to_wavelength
 
         det = cf.Frame2D(name='detector', axes_order=(0, 1))
-        sky = cf.CelestialFrame(name='sky', axes_order=(0, 1),
-                                reference_frame=coord.ICRS())
+        if model.meta.wcs.output_frame.name == 'world':
+            sky = cf.CelestialFrame(name='sky', axes_order=(0, 1),
+                                    reference_frame=coord.ICRS())
+        else:
+            sky = cf.Frame2D(name='sky', axes_order=(0, 1))
```

Comment:
The name is a bit misleading since it's definitely not `sky`. Is something like 
`resampled_+ {model.meta.wcs.output_frame.name}` better?

Reply:
Yes, I thought the naming was only temporary for the purposes of doing the transformation, but now I see that it appears in the meta.wcs.output_frame of the slits in the _s2d file.